charts/super-agent/charts/super-agent-deployment/templates/preinstall-job-register-system-identity.yaml

```diff
@@ -1,22 +1,54 @@
-{{- if .Values.config.auth -}}
-{{- if and .Values.config.auth.enabled .Values.config.auth.secret.create -}}
+{{- if include "newrelic-super-agent.auth.secret.shouldRunJob" . -}}
+{{- /*
+User key secret is used only in the step that create the system identity.
+The secret that is created by the common-library does not allow to add annotations so the secret is removed
+once the installation hook is finished, so I have to add it as a hook.
+*/ -}}
+{{- if not (include "newrelic.common.userKey._customSecretName" .) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    helm.sh/hook: pre-install # TODO we cannot enable auth after installation, we should add pre-upgrade and check if the secret exists
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-weight: "-1010"
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "preinstall-user-key" ) }}
+  namespace: {{ .Release.Namespace }}
+data:
+  {{ include "newrelic.common.userKey.secretKeyName" . }}: {{ include "newrelic.common.userKey._userKey" . | b64enc }}
+{{- end }}
+---
 apiVersion: batch/v1
 kind: Job
 metadata:
   annotations:
     helm.sh/hook: pre-install # TODO we cannot enable auth after installation, we should add pre-upgrade and check if the secret exists
```

Comment:
it would be nice to fix this

Reply:
Done [#e9a9af8](https://github.com/newrelic/helm-charts/pull/1466/commits/e9a9af8741a542fa0cc39ccc41e80b8c85528749)